### ISSUE TITLE
Licensing conflict with xclim

### ISSUE THREAD

**Issue by saschahofmann** [Ouranosinc/lmoments3#12]
As mentioned in [this discussion](https://github.com/Ouranosinc/xclim/discussions/1616#) on the xclim forum. There currently is a licensing conflict with xclim's Apache 2.0 license and lmoments3 GPL 3 license. The easiest way to resolve this conflict would be to change lmoments 3 license to a more permissive license (e.g. like Apache 2.0). I am using this issue to reach to all lmoments3 contributors and ask whether they'd be willing to change lmoments3 license?

The repo mentions 
- [x] @Zeitsperre 
- [x] @aulemahal 
- [x] @faph 
- [x] @huard 
- [x] @monocongo  
- [x] Sam Gillespie @samgillespie
- [ ] @wasquith

**Comment by Zeitsperre** [comment]
@saschahofmann

Just to clarify that @faph is Florenz A.P. Hollebrandse!

**Comment by saschahofmann** [comment]
In the code I can also see these J. R. M. Hosking, William Asquith, Pierre Gérard-Marchant mentioned do you by any chance have contact details for any of them?

**Comment by Zeitsperre** [comment]
Unfortunately, no. I have no idea how to contact them.

**Comment by aulemahal** [comment]
I think those names are associated with the Fortran code, which has a much more permissive licence, no ?

**Comment by saschahofmann** [comment]
They are mentioned in distr.py: https://github.com/Ouranosinc/lmoments3/blob/main/lmoments3/distr.py#L2
But yes the fortran code has a more permissive disclaimer.

**Comment by faph** [comment]
I have no problem with that as far as my own contributions are concerned.

**Comment by huard** [comment]
Good for me as well.

**Comment by Zeitsperre** [comment]
I've already expressed this elsewhere, but just for official purposes, I am comfortable with the licensing changes so long as the other authors are on board!

**Comment by SarahG-579462** [comment]
FYI William Asquith is active on github as @wasquith, for the R code. His code is under GPL v3.0. I'm not sure who provided the translation, it doesn't appear to be in Sam Gillespie's original translation (https://github.com/samgillespie/Python-L-Moments).

Sam Gillespie is also on github @samgillespie, they might be able to be contacted through there, but from what I understand their licensing is more permissive, IANAL, but simply including their license would likely be sufficient.

**Comment by saschahofmann** [comment]
Thanks. Two good resources but yeah seems pretty hard to deduce who needs to give their consent. I added both to the list above for now.

**Comment by samgillespie** [comment]
I'm also comfortable with these licensing changes, I'm happy for this code to be as freely distributable as possible.

**Comment by Zeitsperre** [comment]
I'm going to send an email directly to @monocongo to see if we can get his OK on this. I'll ask if he can weigh in on the issue here.

Unfortunately, I'm still not sure what we can do about gaining @wasquith's approval. Perhaps we could also email him directly?

**Comment by saschahofmann** [comment]
Great. I just emailed @wasquith

**Comment by Zeitsperre** [comment]
Just received an email from @monocongo:

> Hi Trevor,
>
> I'm happy to follow you guys' lead. Please let me know what I can do to help. Kudos on all your hard work around this. I rewrote the L-moments code into Java and Python, more than once, from the original Hoskins Fortran code and it flummoxed me every time. This stuff is hard even without the additional licensing complexity. Godspeed!
>
> --James

All good!

**Comment by aulemahal** [comment]
Due to missing approbation from William Asquith, whose work is cited in the code as the source for part of the implementation, we decided to remove `lmoments3` from the dependencies of xclim.

One can now pass a `lmoments3` object to xclim's functions to retrieve the previous behaviour, see the PR linked above.

We can easily add `lmoments3` as a dependency as soon as we have a positive response from that last person!

**Comment by Zeitsperre** [comment]
Just as an FYI, `xclim` no longer requires (or links to) the `lmoments3` package, but we've changed the way that `scipy` distribution functions are used in order to allow for users to still perform the same analyses, though. This change can always be reverted when the licensing situation changes.

We'll leave this issue open for the time being.

**Comment by aulemahal** [comment]
I just found out that most of the L-moments fortran code is implemented here : https://cran.r-project.org/web/packages/lmom/index.html. This package is (actively) maintained by Hosking himself and is released under the CPL 1.0. 
According to this https://www.tldrlegal.com/license/common-public-license-1-0-cpl-1-0, I think this license would allow sublicensing.

Thus a solution for this issue would be to say that this package is a derivative of that R package, and we could go forward with #14. I think.

**Comment by saschahofmann** [comment]
But is that accurate? My understanding was that this repo is a derivative of this  https://github.com/wasquith/lmomco ?

**Comment by aulemahal** [comment]
You are right, I kinda read our license too fast, it does mention « Additional code » from R's `lmomco`. I have no idea which parts it concerns, so I can't judge on the scale of the issue.
